src/time.cpp

```diff
@@ -58,14 +58,32 @@ static const char *ts2n(time_sink s) {
   return "everything else";
 }
 
-static void pt(double ts[], time_sink s) {
-  if (ts[s]) master_printf("    %21s: %g s\n", ts2n(s), ts[s]);
+static void pt(double mean[], double stddev[], time_sink s) {
+  if (mean[s] != 0) {
+    if (stddev[s] != 0)
+      master_printf("    %21s: %g s +/- %g s\n", ts2n(s), mean[s], stddev[s]);
+    else
+      master_printf("    %21s: %g s\n", ts2n(s), mean[s]);
+  }
 }
 
 void fields::print_times() {
+  double mean[Other + 1], square_times[Other + 1], stddev[Other + 1];
+  int n = count_processors();
+
+  for (int i = 0; i <= Other; ++i)
+    square_times[i] = times_spent[i] * times_spent[i];
+  sum_to_master(times_spent, mean, Other+1);
+  sum_to_master(square_times, stddev, Other+1);
+  for (int i = 0; i <= Other; ++i) {
+    mean[i] /= n;
+    stddev[i] -= n*mean[i]*mean[i];
+    stddev[i] = n == 1 || stddev[i] <= 0 ? 0.0 : sqrt(stddev[i] / (n-1));
```

Comment:
note that this is computing `stddev = ` ∑(xᵢ-x̄)² / (n-1) = (∑xᵢ² - nx̄²) / (n-1)